Test/local data source

Noor-Serry opened the pull request:
# Pull Request Template

## Description
This pull request adds comprehensive unit tests to achieve 100% test coverage for local data source implementations and their related DAOs (except for TvShowDao, which will be refactored before testing).

## Changes Made
✅ Added unit tests for local data source implementations

✅ Wrote tests for DAOs (excluding TvShowDao)

✅ Refactored and migrated tests to JUnit 5

✅ Removed redundant functions that duplicated existing functionality

## Screenshots (if applicable)
Data Source Implementation Coverage:
<img width="916" height="186" alt="image" src="https://github.com/user-attachments/assets/cb0b6fc3-9a63-4180-827b-216448953d0e" />

DAO Test Coverage:
<img width="907" height="743" alt="image" src="https://github.com/user-attachments/assets/614e5aa3-d5fa-4fb5-9a80-b4d9ae9e8097" />


---

## Checklist
Please ensure the following tasks are completed:

- [x] My code follows the code style of this project
- [x] Changes have been tested manually and verified.
- [x] PR includes at most one single feature.

---

## Additional Comments
Add any additional information or context about the pull request here.

Commit:
test: add unit tests for DAO layer

This commit adds a suite of unit tests for the Data Access Object (DAO) layer of the local datasource.

The tests cover the following DAOs:
- `CategoryDaoTest`: Verifies insertion, updating, and retrieval of movie and TV show categories.
- `CountryDaoTest`: Tests insertion, updating, and language-specific retrieval of countries.
- `MovieCategoryInterestDaoTest`: Ensures correct insertion, incrementing, and retrieval of movie category interests.
- `MovieDaoTest`: Validates movie insertion, updates, search entry association, and retrieval by keyword with pagination.
- `TvShowDaoTest`: Tests insertion of TV shows and their search mappings.

Helper functions for creating test data (e.g., `createMovie`, `createTvShow`, `createLocalMovieCategoryDto`) have also been added to facilitate testing.

Commit:
test: migrate localDataSource unit tests to Junit5

This commit migrates all unit tests for the `localDataSource` module from JUnit 4 to JUnit 5.
It also refactors existing tests to improve clarity and maintainability.

Key changes:
- Replaced `@Test` and `@Before` annotations with their JUnit 5 counterparts (`@Test`, `@BeforeEach`).
- Updated assertions to use `Truth` library for better readability.
- Reorganized test classes and methods for better structure.
- Removed unused or redundant test files (`DummyData.kt`, some specific DataSource tests).
- Added new test files for missing components (e.g., `RecentSearchDaoTest.kt`, `TvShowCategoryInterestDaoTest.kt`).
- Ensured test methods follow the Given-When-Then pattern.
- Added utility functions for creating test data (e.g., `createMovie.kt`, `createTvShow.kt`, `createLocalSearchDto.kt`).
- Updated `libs.versions.toml` to include JUnit 5 dependencies and remove JUnit 4.
- Updated `localDatasource/build.gradle.kts` to reflect new testing dependencies.
- Corrected minor logic in `MovieLocalDataSourceImpl`, `RecentSearchLocalDataSourceImpl`, and `TvShowLocalDataSourceImpl`.
- Removed `@Transaction` annotation from `MovieLocalDataSourceImpl` and `TvShowLocalDataSourceImpl` as it's not strictly necessary for the operations.
- Removed `deleteRecentSearchByKeyword` from `RecentSearchLocalSource` and its implementation as it was redundant with `deleteRecentSearchByKeywordAndType`.
- Updated `RecentSearchRepositoryImpl` to use the correct deletion method.

Commit:
chore: remove unused imports

This commit removes unused imports from various Kotlin files.
- `MovieGenre` and `LocalMovieCategoryInterestDto` were removed from `TvShowCategoryInterestDaoTest.kt`.
- `LocalMovieDto` was removed from `MovieDaoTest.kt`.
- `Transaction` was removed from `MovieLocalDataSourceImpl.kt`.

Commit:
Merge branch 'develop' of github.com:Amsterdam-Team/Aflami into test/local-data-source

# Conflicts:
#	gradle/libs.versions.toml
#	localDatasource/src/test/java/com/example/localdatasource/DummyData.kt

eslam-99 commented on gradle/libs.versions.toml:
```diff
@@ -59,6 +67,7 @@ androidx-ui-graphics = { group = "androidx.compose.ui", name = "ui-graphics" }
 androidx-ui-tooling = { group = "androidx.compose.ui", name = "ui-tooling" }
 androidx-ui-tooling-preview = { group = "androidx.compose.ui", name = "ui-tooling-preview" }
 androidx-appcompat = { group = "androidx.appcompat", name = "appcompat", version.ref = "appcompat" }
+junit-jupiter-api-v572 = { module = "org.junit.jupiter:junit-jupiter-api", version.ref = "junitJupiterApiVersion" }
```
Already found in line 91

eslam-99 commented on localDatasource/build.gradle.kts:
```diff
@@ -4,10 +4,16 @@ plugins {
     alias(libs.plugins.ksp)
     alias(libs.plugins.aflami.custom.plugin)
     alias(libs.plugins.kover)
+    id("de.mannodermaus.android-junit5")
```
Please use alias like others

eslam-99 commented on localDatasource/src/androidTest/kotlin/com/example/localdatasource/daos/CategoryDaoTest.kt:
```diff
@@ -0,0 +1,175 @@
+    @Test
+    fun upsertAllMovieCategories_shouldUpdateMovieCategories_whenAlreadyStored() = runTest {
+        // Given
+        val initial = listOf(
```
Squad agreed to extract data to companion object or another file, you can follow what applied in this layer

Noor-Serry replied:
already I did that there are utils file that help me to create objects

eslam-99 replied:
I mean to predefine some lists and data not just use
```createLocalTvShowCategoryDto(categoryId = 2, name = "Old", storedLanguage = "en")```
I saw some tests predefined some data as I told you

Noor-Serry replied:
i did not need it

Commit:
chore: update testing dependencies

This commit updates the following testing dependencies:
- JUnit Jupiter API to version 5.10.3
- Truth to version 1.4.4

The `localDatasource/build.gradle.kts` file has been updated to use the new version catalog aliases for these dependencies. The old version references have been removed from `gradle/libs.versions.toml`.

Commit:
Merge branch 'develop' of github.com:Amsterdam-Team/Aflami into test/local-data-source

# Conflicts:
#	gradle/libs.versions.toml
#	localDatasource/build.gradle.kts
#	localDatasource/src/main/java/com/example/localdatasource/roomDataBase/datasource/MovieLocalDataSourceImpl.kt
#	localDatasource/src/main/java/com/example/localdatasource/roomDataBase/datasource/TvShowLocalDataSourceImpl.kt
#	localDatasource/src/test/java/com/example/localdatasource/TvShowLocalDataSourceImplTest.kt

Commit:
test: update tests for local data sources and add app data store tests

This commit includes the following changes:

- Updates tests for `MovieLocalDataSourceImplTest` and `TvShowLocalDataSourceImplTest` to reflect changes in method signatures and data handling.
- Specifically, tests for `getMoviesBySearchKeywordSortedByInterest`, `addMoviesBySearchData`, `incrementGenreInterest`, and `addMovieWithCategories` (for movies) and `getTvShowsBySearchKeywordSortedByInterest`, `addTvShows`, and `incrementGenreInterest` (for TV shows) have been modified.
- Test utility functions `createMovie` and `createTvShow` now use `releaseDate` and `airDate` (respectively) of type `LocalDate` instead of `productionYear`.
- Adds new test utility functions `createLocalMovieCategoryDto` and `createLocalTvShowCategoryDto`.
- Introduces new tests for `AppDataStorePreferences` to verify `getSessionType` and `setSessionType` functionality.
- Removes the `@Transaction` annotation from `addTvShows` in `TvShowLocalDataSourceImpl` as it's not strictly necessary for the operations performed.
- Updates the `shaded` library version in `libs.versions.toml`.
